libqtile/widget/wlan.py

```diff
@@ -41,6 +43,26 @@ def get_status(interface_name):
     return essid, quality
 
 
+def get_private_ip(interface_name):
+    try:
+        result = subprocess.run(
+            ["ip", "-brief", "addr", "show", "dev", interface_name],
+            capture_output=True,
+            text=True,
+            check=True,
+        )
+        output = result.stdout.strip()
+        parts = output.split()
+        if len(parts) > 2 and parts[1] == "UP":
+            ip_address = parts[2].split("/")[0]
+            if ":" not in ip_address:
+                return ip_address
+    except Exception:
+        logger.exception(f"Couldn't get the IP for {interface_name}:")
+
+    return "N/A"
```

Comment:
This `try... except...` block is too big and it's generally bad practice to catch `Exception` as you would swallow unexpected scenarios.

The only thing that would throw an exception here is the `subprocess.run` call. Everything else should be handled by your logic.

```suggestion
def get_private_ip(interface_name):
    try:
        result = subprocess.run(
            ["ip", "-brief", "addr", "show", "dev", interface_name],
            capture_output=True,
            text=True,
            check=True,
        )
    except subprocess.CalledProcessError:
        logger.exception(f"Couldn't get the IP for {interface_name}:")
        return "N/A"

    output = result.stdout.strip()
    parts = output.split()
    if len(parts) > 2 and parts[1] == "UP":
        ip_address = parts[2].split("/")[0]
        if ":" not in ip_address:
            return ip_address

    return "N/A"
```